src/java.desktop/share/classes/javax/swing/plaf/basic/BasicScrollBarUI.java

```diff
@@ -1608,6 +1608,25 @@ public ScrollListener(int dir, boolean block)   {
 
         /** {@inheritDoc} */
         public void actionPerformed(ActionEvent e) {
+            // If frame is disabled and time is started in mousePressed
+            // and mouseReleased is not called, then timer will not be stopped
+            // Stop the timer if frame is disabled
+            Component parent = scrollbar.getParent();
+            do {
+                if (parent instanceof javax.swing.JFrame par) {
+                    if (!par.isEnabled()) {
+                        ((Timer)e.getSource()).stop();
```

Comment:
```suggestion
                        ((Timer) e.getSource()).stop();
```

Reply:
other places in same method has same spacing...they can be done under one cleanup issue..